cmd/tether/attach.go

```diff
@@ -140,26 +179,82 @@ func (t *attachServerSSH) start() error {
 	}
 	t.sshConfig.AddHostKey(pkey)
 
-	t.enabled = true
+	t.Enabled()
 	go t.run()
 
 	return nil
 }
 
 // stop is not thread safe with start
-func (t *attachServerSSH) stop() {
+func (t *attachServerSSH) stop() error {
 	defer trace.End(trace.Begin("stop attach server"))
 
-	if t == nil || !t.enabled {
-		return
+	if t == nil {
+		err := fmt.Errorf("attach server is not configured")
+		log.Error(err)
+		return err
 	}
 
-	t.enabled = false
-	conn := t.conn
-	t.conn = nil
+	if !t.IsEnabled() {
+		err := fmt.Errorf("attach server is not enabled")
+		log.Error(err)
+		return err
+	}
+
+	log.Debugf("Setting enabled to false")
+	t.Disabled()
+
+	// This context is used by backchannel and we need to cancel it before
+	// trying to obtain the following lock so that backchannel releases it.
```

Comment:
It's not that backchannel holds this lock, but that this lock is held by backchannel's caller and not released until backchannel returns.